pcl_conversions/CMakeLists.txt

```diff
@@ -1,44 +1,63 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.5)
 project(pcl_conversions)
 
-find_package(catkin REQUIRED COMPONENTS)
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
-find_package(PCL REQUIRED COMPONENTS common io)
+find_package(ament_cmake REQUIRED)
+find_package(Boost REQUIRED)
 find_package(Eigen3 REQUIRED)
+find_package(message_filters REQUIRED)
+find_package(PCL REQUIRED QUIET COMPONENTS common io)
+find_package(pcl_msgs REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(std_msgs REQUIRED)
 
-# There is a bug in the Ubuntu Artful (17.10) version of the VTK package,
-# where it includes /usr/include/*-linux-gnu/freetype2 in the include
-# directories (which doesn't exist).  This filters down to the PCL_INCLUDE_DIRS,
-# and causes downstream projects trying to use these libraries to fail to
-# configure properly.  Here we remove those bogus entries so that downstream
-# consumers of this package succeed.
-if(NOT "${PCL_INCLUDE_DIRS}" STREQUAL "")
-  foreach(item ${PCL_INCLUDE_DIRS})
-    string(REGEX MATCH "/usr/include/.*-linux-gnu/freetype2" item ${item})
-    if(item)
-      list(REMOVE_ITEM PCL_INCLUDE_DIRS ${item})
-    endif()
-  endforeach()
-endif()
 
-catkin_package(
-  INCLUDE_DIRS include
-  CATKIN_DEPENDS pcl_msgs roscpp sensor_msgs std_msgs
-  DEPENDS EIGEN3 PCL
+set(dependencies
+  pcl_msgs
+  rclcpp
+  sensor_msgs
+  std_msgs
+)
+
+set(includes
+  include
+  ${Eigen_INCLUDE_DIRS}
+  ${message_filters_INCLUDE_DIRS}
+  ${PCL_COMMON_INCLUDE_DIRS}
+  ${pcl_msgs_INCLUDE_DIRS}
+  ${rclcpp_INCLUDE_DIRS}
+)
+
+include_directories(
+  ${includes}
 )
 
 install(DIRECTORY include/${PROJECT_NAME}/
-  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+  DESTINATION include/${PROJECT_NAME}/
 )
 
-if(CATKIN_ENABLE_TESTING)
-  find_package(catkin REQUIRED COMPONENTS roscpp pcl_msgs sensor_msgs std_msgs)
-  include_directories(
-    include
-    ${catkin_INCLUDE_DIRS}
-    ${PCL_INCLUDE_DIRS}
-    ${EIGEN3_INCLUDE_DIRS})
+# Add gtest based cpp test target
+if(BUILD_TESTING)
+  find_package(ament_cmake_gtest REQUIRED)
 
-  catkin_add_gtest(test_pcl_conversions test/test_pcl_conversions.cpp)
-  target_link_libraries(test_pcl_conversions ${catkin_LIBRARIES})
+  ament_add_gtest(${PROJECT_NAME}-test test/test_pcl_conversions.cpp)
+  ament_target_dependencies(${PROJECT_NAME}-test
+    ${dependencies}
+  )
+  target_link_libraries(${PROJECT_NAME}-test ${Boost_LIBRARIES})
 endif()
+
+ament_export_include_directories(
+  ${includes}
+)
+ament_export_dependencies()
```

Comment:
You didnt include any depends to the export depends `${dependencies}`

Reply:
oops, fixed.